src/foraging_gui/GenerateMetadata.py

```diff
@@ -0,0 +1,988 @@
+            grant_number=self.Obj['meta_data_dialog']['session_metadata']['GrantNumber'],
+            fundee=self.Obj['meta_data_dialog']['session_metadata']['Fundee'],
+        )]
+                
+    def _get_platform(self):
+        '''
+        Get the platform name. This should be improved in the future.
+        '''
+        if self.box_type == 'Ephys':
+            self.platform = Platform.ECEPHYS
+        elif self.box_type == 'Behavior':
+            self.platform = Platform.BEHAVIOR
+        else:
+            self.platform = ''
+
+    def _get_session_time(self):
+        '''
+        Get the session start and session end time
+        '''
+        # priority behavior_streams>high_speed_camera_streams>ephys_streams>ophys_streams
+        if self.behavior_streams!=[]:
+            self.session_start_time = self.behavior_streams[0].stream_start_time
+            self.session_end_time = self.behavior_streams[0].stream_end_time
+        elif self.high_speed_camera_streams!=[]:
+            self.session_start_time = self.high_speed_camera_streams[0].stream_start_time
+            self.session_end_time = self.high_speed_camera_streams[0].stream_end_time
+        elif self.ephys_streams!=[]:
+            self.session_start_time = self.ephys_streams[0].stream_start_time
+            self.session_end_time = self.ephys_streams[0].stream_end_time
+        elif self.ophys_streams!=[]:
+            self.session_start_time = self.ophys_streams[0].stream_start_time
+            self.session_end_time = self.ophys_streams[0].stream_end_time
+        else:
+            self.session_start_time = ''
+            self.session_end_time = ''
+
+    def _get_modality(self):
+        '''
+        Get all the modalities used in the session
+        '''
+        self.modality = []
+        if self.behavior_streams!=[]:
+            self.modality.append(Modality.BEHAVIOR)
+        if self.ephys_streams!=[]:
+            self.modality.append(Modality.ECEPHYS)
+        if self.ophys_streams!=[]:
+            self.modality.append(Modality.FIB)
+        if self.high_speed_camera_streams!=[]:
+            self.modality.append(Modality.BEHAVIOR_VIDEOS)
+        
+    def _get_investigators(self):
+        '''
+        Get investigators
+        '''
+        self.investigators=[]
+        investigators=self.Obj['meta_data_dialog']['session_metadata']['Investigators'].split(',')
+        for investigator in investigators:
+            if investigator != '':
+                self.investigators.append(PIDName(name=investigator, registry=self.orcid))
+
+    def _save_rig_metadata(self):
+        '''
+        Save the rig metadata to the MetadataFolder
+        '''
+        if self.Obj['meta_data_dialog']['rig_metadata_file']=='' or self.Obj['MetadataFolder']=='':
+            return
+        
+        rig_metadata_full_path=os.path.join(self.Obj['MetadataFolder'],self.Obj['meta_data_dialog']['rig_metadata_file'])
```

Comment:
We aren't updating the rig metadata, we are just loading it from file right? So why not just copy the file, instead of writing a new one? Less chance of a bug being created that way

Reply:
I don't update the rig metadata. I stored a copy of the rig metadata to the behavior json, which means we can generate the session and rig metadata just from the behavior json, and the rig metadata is not necessarily from an outside file.